src/Combinatorics/Matroids/ChowRings.jl

```diff
@@ -0,0 +1,211 @@
+export chow_ring, augmented_chow_ring, select
+
+@doc Markdown.doc"""
+The Chow ring of a matroid, optional also with the simplical generators.
+
+See AHK18 (@cite) and BES21 (@cite) 
+
+
+# Examples
+The following computes the chow ring of the Fano matroid.
+```jldoctest
+julia> M = fano_matroid();
+
+julia> R = chow_ring(M);
+
+julia> R[1]*R[8]
+-x_{3,4,7}^2
+```
+
+# Examples
+The following computes the chow ring of the Fano matroid including variables for the simplicial generators.
+```jldoctest
+julia> M = fano_matroid();
+
+julia> R = chow_ring(M, extended=true);
+
+julia> f = R[22] + R[8] - R[29]
+x_{1,2,3} + h_{1,2,3} - h_{1,2,3,4,5,6,7}
+
+julia> f==0
+true
+```
+"""
+function chow_ring(M::Matroid; extended=false)
+        Flats = flats(M)
+        number_flats = length(Flats)
+        n = length(M.groundset)
+        if !is_loopless(M)
+                error("Matroid has loops")
+        end
+        if number_flats<2
+                error("matroid has to few flats")
+        end
+        proper_flats = Flats[2:number_flats-1]
+        var_names = [replace(string("x_",S), "["=>"{", "]"=>"}", ", "=>",") for S in proper_flats] # create variable names, indexed by the proper flats of M
+        if extended
+                var_names = [var_names; [replace(string("h_",S), "["=>"{", "]"=>"}", ", "=>",") for S in [proper_flats;[Flats[number_flats]]]]] #addeds the variables for the simplicial generators
+        end
+
+        ring, vars = GradedPolynomialRing(QQ, var_names)
+        #ring, vars = PolynomialRing(ZZ, var_names) $which ring do we want here
+        
+	I = linear_relations(ring, proper_flats, vars, M)
+        J = quadratic_relations(ring, proper_flats, vars)
+        Ex = []
+        if extended
+                Ex = relations_extended_ring(ring, proper_flats, vars)
+        end
+        chow_modulus = ideal(ring, vcat(I, J, Ex))
+        #chow_ring = ResidueRing(ring, chow_modulus)# is this a better choice?
+        chow_ring, projection = quo(ring, chow_modulus)
+        return chow_ring
+end
+
+function linear_relations(ring, proper_flats, vars, M)
+        alpha = 0
```

Comment:
Your use of `alpha` is not type stable. I am guessing you want this:
```suggestion
        alpha = zero(R)
```